src/engine/plugins/tile.cpp

```diff
@@ -380,226 +387,218 @@ Status TilePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFacad
-                // shortcuts
-                if (smaller_edge_id != SPECIAL_EDGEID)
-                {
-                    // Check to see if it was a shortcut edge we found.  This can happen
-                    // when exactly?  Anyway, unpack it and get the first "real" edgedata
-                    // out of it, which should represent the first hop, which is the one
-                    // we want to find the turn.
-                    const auto &data = [this,
-                                        &facade,
-                                        smaller_edge_id,
-                                        source_ebn,
-                                        target_ebn,
-                                        &unpacked_shortcut]() {
-                        const auto inner_data = facade->GetEdgeData(smaller_edge_id);
-                        if (inner_data.shortcut)
+                        // If no edge was found, it means that there's no connection between these
+                        // nodes, due to oneways or turn restrictions. Given the edge-based-nodes
+                        // that we're examining here, we *should* only find directly-connected
+                        // edges, not shortcuts
+                        if (smaller_edge_id != SPECIAL_EDGEID)
                         {
-                            unpacked_shortcut.clear();
-                            UnpackEdgeToEdges(
-                                *facade, source_ebn.first, target_ebn, unpacked_shortcut);
-                            return unpacked_shortcut.front();
+                            // Check to see if it was a shortcut edge we found.  This can happen
+                            // when exactly?  Anyway, unpack it and get the first "real" edgedata
+                            // out of it, which should represent the first hop, which is the one
+                            // we want to find the turn.
+                            const auto &data = [this,
+                                                &facade,
+                                                smaller_edge_id,
+                                                firstedge,
+                                                secondedge,
+                                                &unpacked_shortcut]() {
+                                const auto inner_data = facade->GetEdgeData(smaller_edge_id);
+                                if (inner_data.shortcut)
+                                {
+                                    unpacked_shortcut.clear();
+                                    UnpackEdgeToEdges(*facade,
+                                                      firstedge.edge_based_node_id,
+                                                      secondedge.edge_based_node_id,
+                                                      unpacked_shortcut);
+                                    return unpacked_shortcut.front();
```

Comment:
if we only care about the very first node, it might give some performance if we implement a `left sided` unpack that does not recurse in both directions. Should switch from `O(n)` to `O(log(n))` on average, assuming a nice and balanced shortcut creation. (Worst case will stay `O(n)`)